src/main/kotlin/org/rust/cargo/project/RsToolchainPathChoosingComboBox.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project
+
+import com.intellij.openapi.fileChooser.FileChooser
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory
+import com.intellij.openapi.ui.ComboBoxWithWidePopup
+import com.intellij.openapi.ui.ComponentWithBrowseButton
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.ComboboxSpeedSearch
+import com.intellij.ui.components.fields.ExtendableTextComponent
+import com.intellij.ui.components.fields.ExtendableTextField
+import org.rust.openapiext.pathAsPath
+import java.nio.file.Path
+import javax.swing.plaf.basic.BasicComboBoxEditor
+
+/**
+ * A combobox with browse button for choosing a path to a toolchain, also capable of showing progress indicator.
+ * To toggle progress indicator visibility use [setBusy] method.
+ */
+class RsToolchainPathChoosingComboBox : ComponentWithBrowseButton<ComboBoxWithWidePopup<Path>>(ComboBoxWithWidePopup(), null) {
+    private val editor: BasicComboBoxEditor = object : BasicComboBoxEditor() {
+        override fun createEditorComponent() = ExtendableTextField().apply { isEditable = false }
```

Comment:
But you can do it via this text field (selected):

<img width="539" alt="Screenshot 2021-06-09 at 18 44 21" src="https://user-images.githubusercontent.com/6079006/121386504-ba001180-c952-11eb-8d29-ee026ea71880.png">

I don't know how to subscribe to text change event for `RsToolchainPathChoosingComboBox`, it doesn't have `document` field.

Reply:
> But you can do it via this text field (selected):

Yep, but it's less convenient since it requires additional action and not so obvious at all

> I don't know how to subscribe to text change event for RsToolchainPathChoosingComboBox, it doesn't have document field.

I suppose you should use `javax.swing.text.JTextComponent#getDocument` from `ExtendableTextField`